roadmap/implementers-guide/src/node/approval/approval-voting.md

```diff
@@ -287,37 +288,78 @@ On receiving an `ApprovedAncestor(Hash, BlockNumber, response_channel)`:
+
+Likewise, when considering how many tranches to take, the no-show depth should be used to apply a depth-discount or clock drift to the `tranche_now`.
+
+**Procedure**
+
+  * Start with `depth = 0`.
+  * Set a clock drift of `depth * no_show_duration`
+  * Take tranches up to `tranche_now - clock_drift` until all needed assignments are met. Start with `depth = 0`.
+  * Keep track of the `next_no_show` according to the clock drift, as we go.
+  * If running out of tranches before then, return `Pending { considered, next_no_show, maximum_broadcast, clock_drift }`
+  * If there are no no-shows, return `Exact { needed, tolerated_missing, next_no_show }`
+  * `maximum_broadcast` is either `DelayTranche::max_value()` at tranche 0 or otherwise by the last considered tranche + the number of uncovered no-shows at this point.
+  * If there are no-shows, return to the beginning, incrementing `depth` and attempting to cover the number of no-shows. Each no-show must be covered by a non-empty tranche, which are tranches that have at least one assignment. Each non-empty tranche covers exactly one no-show.
+  * If at any point, it seems that all validators are required, do an early return with `RequiredTranches::All` which indicates that everyone should broadcast.
 
 #### Check Approval
   * Check whether a candidate is approved under a particular block.
   * Requires `(block_entry, candidate_entry, approval_entry, n_tranches)`
   * If `n_tranches` is `RequiredTranches::Pending`, return false
   * If `n_tranches` is `RequiredTranches::All`,  then we return `3 * n_approvals > 2 * n_validators`.
-  * If `n_tranches` is `RequiredTranches::Exact(tranche, no_shows)`, then we return whether all assigned validators up to `tranche` less `no_shows` have approved. e.g. if we had 5 tranches and 1 no-show, we would accept all validators in tranches 0..=5 except for 1 approving. In that example, we also accept all validators in tranches 0..=5 approving, but that would indicate that the `RequiredTranches` value was incorrectly constructed, so it is not realistic. If there are more missing approvals than there are no-shows, that indicates that there are some assignments which are not yet no-shows, but may become no-shows.
+  * If `n_tranches` is `RequiredTranches::Exact { tranche, tolerated_missing, .. }`, then we return whether all assigned validators up to `tranche` less `tolerated_missing` have approved. e.g. if we had 5 tranches and 1 no-show, we would accept all validators in tranches 0..=5 except for 1 approving. In that example, we also accept all validators in tranches 0..=5 approving, but that would indicate that the `RequiredTranches` value was incorrectly constructed, so it is not realistic. If there are more missing approvals than there are no-shows, that indicates that there are some assignments which are not yet no-shows, but may become no-shows.
```

Comment:
I'm slightly confused by the e.g.  here using no-show the way it does, but actually `tolerated_missing` sems pretty clear: no-shows become `tolerated_missing` when we replace them.